scopes/scope/scope/scope.main.runtime.ts

```diff
@@ -351,8 +353,39 @@ export class ScopeMain implements ComponentFactory {
   private localAspects: string[] = [];
 
   async loadAspects(ids: string[], throwOnError = false): Promise<void> {
+    const scopeManifests = await this.getManifestsGraphRecursively(ids, throwOnError);
+    await this.aspectLoader.loadExtensionsByManifests(scopeManifests);
+  }
+
+  async getManifestsGraphRecursively(ids: string[], throwOnError = false): Promise<ManifestOrAspect[]> {
+    if (!ids.length) return [];
+    const components = await this.getNonLoadedAspects(ids);
```

Comment:
This loadedAspects will only be updated at the end, after passing everything to harmony.
Which means you will do the require multiple times in some cases.
And means you will have the same manifest a few times (not sure how harmony will handle this. it might throw an error)

Reply:
@GiladShoham , yes, that's a good point.
With the `visited` approach, I guess it won't happen, as it never loading the same id twice.